python/GafferSceneTest/InstancerTest.py

```diff
@@ -2630,6 +2630,50 @@ def testRootPerVertexWithEmptyPoints( self ) :
 		self.assertEqual( instancer["out"].childNames( "/points/instances" ), IECore.InternedStringVectorData() )
 		self.assertSceneValid( instancer["out"] )
 
+	def testPurpose( self ):
+
+		points = IECoreScene.PointsPrimitive(
+					IECore.V3fVectorData(
+						[ imath.V3f( i, 0, 0 ) for i in range( 100 ) ]
+					)
+				)
+
+		points["intVar"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, IECore.IntVectorData(
+						[ i for i in range( 100 ) ]
+					) )
+		pointsSource = GafferScene.ObjectToScene()
+		pointsSource["name"].setValue( "points" )
+		pointsSource["object"].setValue( points )
+
+		sphere = GafferScene.Sphere()
+
+		sphereFilter = GafferScene.PathFilter()
+		sphereFilter["paths"].setValue( IECore.StringVectorData( [ '/*' ] ) )
+
+		purposeAttr = GafferScene.CustomAttributes()
+		purposeAttr["in"].setInput( sphere["out"] )
+		purposeAttr["filter"].setInput( sphereFilter["out"] )
+		purposeAttr["attributes"].addChild( Gaffer.NameValuePlug( "usd:purpose", "default" ) )
+		purposeAttr["expression"] = Gaffer.Expression()
+		purposeAttr["expression"].setExpression( 'parent["attributes"]["NameValuePlug"]["value"] = "default" if context.get("intVar", 1 ) % 2 else "invalid"', "python" )
```

Comment:
I suppose it will mean using a StandardOptions node to choose what purposes to render, but could we use a real purpose rather than `"invalid"` here?